src/main/java/org/wildfly/security/http/impl/FormAuthenticationMechanism.java

```diff
@@ -334,7 +335,7 @@ private void sendPage(String page, HttpServerRequest request, HttpServerResponse
                         while ((length = pageStream.read(content)) > 0) {
                             responseStream.write(content, 0, length);
                         }
-
+                        response.setStatusCode(OK);
```

Comment:
This line illustrates my previous comment, by changing the framework a mechanism is forced to change but mechanisms also exist outside of the WildFly Elytron project as this is a supported SPI.

Reply:
This default is used only here, in FORM mechanism, which, as was discussed later regarding HttpMechanism interface, technically is not HTTP mechanism.

Reply:
FORM authentication is something we need to be able to support with this framework in conjunction with the other authentication mechanisms, at this stage as the SPI has been released in a .Final of WildFly Elytron and WildFly it is always possible other custom mechanism implementations out there are implemented based on the current behaviour of the framework.